realm/realm-library/src/main/java/io/realm/RealmChangeListener.java

```diff
@@ -17,7 +17,13 @@
 package io.realm;
 
 /**
- * Using RealmChangeListener, it is possible to be notified when a Realm instance has been updated.
+ * RealmChangeListener can be registered with a {@link Realm}, {@link RealmResults} or {@link RealmObject}
+ * to receive a notification about updates.
+ * <p>
+ * When registered against a {@code Realm} you'll get notified when a Realm instance has been updated.
+ * When registered against a {@code RealmResults} or {@code RealmObject} you'll get notified only when the
```

Comment:
That is not entirely true (yet). Perhaps we should add a caveat at the end about that?